### PR TITLE
Mask credentials on urls and add timeout to command

### PR DESCRIPTION
**Pull request by sverdlov93**

### CODE REVIEW

**Review by yahavi — CHANGES_REQUESTED** [review]
Looks good!
Just change "arg" to something else like "line".

File: src/main/java/org/jfrog/bamboo/utils/ExecutableRunner.java [hunk 1]
```diff
@@ -19,7 +22,7 @@ public class ExecutableRunner {
     private final Map<String, String> envs;
     private final BuildLog buildLog;
     private final List<String> secrets;
-
+    private static final int COMMAND_TIMEOUT = 45; // minutes
```
Comment (yahavi) [inline src/main/java/org/jfrog/bamboo/utils/ExecutableRunner.java:25]: Let's make it 60 minutes

File: src/main/java/org/jfrog/bamboo/utils/ExecutableRunner.java [hunk 2]
```diff
@@ -58,35 +61,55 @@ public int run(List<String> commandArgs) throws IOException, InterruptedExceptio
...
-                buildLog.info(maskSecrets(line));
+
+        Thread outputReader = new Thread(() -> {
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    buildLog.info(maskSecrets(line));
+                }
+            } catch (IOException e) {
+                throw new RuntimeException(e);
```
Comment (yahavi) [inline src/main/java/org/jfrog/bamboo/utils/ExecutableRunner.java:72]: Is it possible to include the error stream for logging errors as well?

**Review by yahavi — CHANGES_REQUESTED** [review]

File: src/main/java/org/jfrog/bamboo/utils/ExecutableRunner.java (hunk 1 above)
Comment (yahavi) [reply src/main/java/org/jfrog/bamboo/utils/ExecutableRunner.java:25]: Please also make sure to put all comments above the line of code.

File: src/main/java/org/jfrog/bamboo/utils/ExecutableRunner.java (hunk 2 above)
Comment (yahavi) [inline src/main/java/org/jfrog/bamboo/utils/ExecutableRunner.java:72]: I looked in the code and I'm not convinced that the RuntimeException is caught and logged somewhere in the code. Let's at least log the root cause here.

Comment (sverdlov93) [reply src/main/java/org/jfrog/bamboo/utils/ExecutableRunner.java:72]: .redirectErrorStream(true); already do it @yahavi

**Pull request merged**